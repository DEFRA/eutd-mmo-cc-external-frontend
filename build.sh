#/bin/bash
curl -O https://nodejs.org/dist/v12.18.4/node-v12.18.4-linux-x64.tar.xz
tar -xf node-v12.18.4-linux-x64.tar.xz --directory /tmp
export PATH="/tmp/node-v12.18.4-linux-x64/bin:$PATH"
rm node-v12.18.4-linux-x64.tar.xz
npm ci
npm run test-vstack
rm -rf /tmp/node-v12.18.4-linux-x64