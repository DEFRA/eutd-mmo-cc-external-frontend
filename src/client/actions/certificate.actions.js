import { isError, showFullPageError } from './index';

export const GET_SUMMARY_DOCUMENT = 'GET_SUMMARY_DOCUMENT';
export const SHOW_SUMMARY_DOCUMENT_ERRORS = 'SHOW_SUMMARY_DOCUMENT_ERRORS';
export const GET_SUMMARY_UNAUTHORISED = 'GET_SUMMARY_DOCUMENT_UNAUTHORISED';

export function showSummaryCertificateErrors(data) {
  return { type: SHOW_SUMMARY_DOCUMENT_ERRORS, payload: { validationErrors: data } };
}

export const getSummaryCertificate = (journey, documentNumber = undefined) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? { headers: { documentnumber: documentNumber } }
    : {};

  await orchestrationApi.get(`/certificate/${journey}`, config)
    .then(
      res => {
        dispatch({ type: GET_SUMMARY_DOCUMENT, payload: res.data });
      },
      error => {
        if (isError(error)) {
          dispatch(showFullPageError());
        } else {
          dispatch(unauthorised());
        }
      });

    function unauthorised() { return { type: GET_SUMMARY_UNAUTHORISED }; }
};