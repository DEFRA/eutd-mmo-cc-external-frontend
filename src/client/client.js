// Startup point for the client side application
import 'babel-polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import {Router, withRouter} from 'react-router-dom';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import axios from 'axios';
import Routes from './Routes';
import reducers from './reducers';
import { createBrowserHistory } from 'history';

const composeEnhancers = process.env.NODE_ENV === 'development' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__  ?
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose;
  

const paramSerializerFn = params => {
  let output = '';
  Object.keys(params).forEach((k) => {
    output+= `${k}=${params[k]}&`;
  });
  return encodeURI(output);
};

const axiosResponseInterceptorSuccessFn = response => response;
const axiosResponseInterceptorErrorFn = error => {
  if (error.response.status === 599) {
    // Let's check if this is because we're logged out
    const empPromise = new Promise(() => {});
    const searchParam = error.request.responseURL.slice(error.request.responseURL.search(/[?]/g));
    let searchParams = new URLSearchParams(searchParam);
    if(searchParams.get('notLoggedInErr') === 'yes') {
      window.location = '/logout?backToPath=/server-logout';
      return empPromise;
    }
    // It's always a timeout...
    window.location = '/timed-out';
    return new Promise(() => {});
  } else {
    return new Promise((_,rej) => rej(error));
  }
};

const axiosInstanceForOrchService = axios.create({
  baseURL: '/orchestration/api/v1/',
  paramsSerializer: paramSerializerFn,
  timeout: window.INITIAL_STATE.config.axiosTimeout
});

axiosInstanceForOrchService.interceptors.response.use(axiosResponseInterceptorSuccessFn, axiosResponseInterceptorErrorFn);

const axiosInstanceForRefService = axios.create({
  baseURL: '/reference/api/v1/',
  paramsSerializer: paramSerializerFn,
  timeout: window.INITIAL_STATE.config.axiosTimeout
});

axiosInstanceForRefService.interceptors.response.use(axiosResponseInterceptorSuccessFn, axiosResponseInterceptorErrorFn);

const axiosInstanceForDynamixService = axios.create({
  baseURL: '/dynamix/',
  paramsSerializer: paramSerializerFn,
  timeout: window.INITIAL_STATE.config.axiosTimeout
});

axiosInstanceForDynamixService.interceptors.response.use(axiosResponseInterceptorSuccessFn, axiosResponseInterceptorErrorFn);

const store = createStore(
  reducers,
  window.INITIAL_STATE,
  composeEnhancers(applyMiddleware(
    // The difference between orch/ref service api is not significant as they both go through proxy
    // However it is significant in the server side store. To stick with same action creators we're
    // passing down same axios instance to both orchestration and reference service api .
    thunk.withExtraArgument( {
      orchestrationApi: axiosInstanceForOrchService,
      referenceServiceApi: axiosInstanceForRefService,
      dynamixApi: axiosInstanceForDynamixService } )
  ))
);

const ScrollToTop = withRouter( class ScrollToTop extends React.Component {
  componentDidMount() {
  }

  render() {
    return this.props.children;
  }
});

const history = createBrowserHistory();


ReactDOM.hydrate(
  <Provider store={store}>
    <Router history={history}>
      <ScrollToTop>
        <div>{renderRoutes(Routes)}</div>
      </ScrollToTop>
    </Router>
  </Provider>,
  document.querySelector('#root')
);
