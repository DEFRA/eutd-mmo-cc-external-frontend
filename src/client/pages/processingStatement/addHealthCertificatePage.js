import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import {
  Main,
  BackLink,
  InputField,
  Header,
  GridRow,
  GridCol
} from 'govuk-react';

import {
  getProcessingStatementFromRedis,
  saveProcessingStatement,
  saveProcessingStatementToRedis,
  clearProcessingStatement
} from '../../actions';

import PageTemplateWrapper from '../../components/PageTemplateWrapper';
import {
  toGovukErrors,
  onHandleErrorClick,
  scrollToErrorIsland
} from '../utils';
import ErrorIsland from '../../components/elements/ErrorIsland';
import PageTitle from '../../components/PageTitle';
import Form from '../../components/elements/Form';
import DateFieldWithPicker from '../../components/DateFieldWithPicker';
import {withRouter} from 'react-router-dom';
import HelpLink from '../../components/HelpLink';
import SaveAsDraftButton from '../../components/SaveAsDraftButton';
import ContinueButton from '../../components/elements/ContinueButton';
import { withTranslation } from 'react-i18next';
import BackToProgressLink from '../../components/BackToProgressLink';

class AddHealthCertificatePage extends React.Component {
  state = {
    certificateDateString: '',
  };

  async save(nextUri) {
    const { saveAsDraftUri } = this.props.route;
    const { documentNumber } = this.props.match.params;
    const isHealthCertificateSavedAsDraft = nextUri === saveAsDraftUri;
    const processingStatement = await this.props.saveToRedis({
        data: this.props.processingStatement || {},
        currentUrl: `/create-processing-statement/${documentNumber}/add-health-certificate`,
        saveAsDraft: isHealthCertificateSavedAsDraft,
        saveToRedisIfErrors: true,
        documentNumber: documentNumber
      });

    if (!isHealthCertificateSavedAsDraft && !_.isEmpty(processingStatement.errors)) {
      scrollToErrorIsland();
      return;
    }

    this.props.history.push(nextUri.replace(':documentNumber', documentNumber));
  }

  async componentDidMount() {
    await this.props.getFromRedis(this.props.match.params.documentNumber);

     this.setState({
      certificateDateString: this.props.processingStatement.healthCertificateDate,
    });
  }

  isUnauthorised() {
    const { processingStatement } = this.props;
    return processingStatement.unauthorised === true;
  }

  unauthorised() {
    this.props.history.push('/forbidden');
  }

  componentWillUnmount() {
    this.props.clear();
  }

  componentDidUpdate() {
    if (this.isUnauthorised()) {
      this.unauthorised();
    }
  }

  onChange(e) {
    const processingStatement = _.cloneDeep(this.props.processingStatement || {});
    processingStatement[e.target.name] = e.target.value;
    this.props.save(processingStatement);
  }

  goBack(e, page) {
    e.preventDefault();
    this.props.history.push(page);
  }

  onSaveAsDraft = async (e) => {
    e.preventDefault();
    const { saveAsDraftUri } = this.props.route;
    await this.save(saveAsDraftUri);
  }

  render() {
    const { processingStatement = {}, history, route, match, t } = this.props;
    const documentNumber = match.params.documentNumber;
    const { certificateDateString } = this.state;
    const { previousUri, path, saveAsDraftUri, nextUri, journey, progressUri, title } = route;

    const errors = (processingStatement.errorsUrl === history.location.pathname && processingStatement.errors) ? processingStatement.errors : {}; // only display errors for this URL
    const saveAsDraftFormActionUrl = `/orchestration/api/v1/processingStatement/saveAndValidate?c=${path}&n=${nextUri}&saveAsDraftUrl=${saveAsDraftUri}`;

    const backUrl = previousUri.replace(':documentNumber', documentNumber);

    return (
      <Main className="processing-statement">
        <PageTitle title={`${!_.isEmpty(errors) ? 'Error: ' : ''}${t('psAddHealthCertificateHeader')} - ${t(title)}`} />
        <ErrorIsland errors={toGovukErrors(errors)} onHandleErrorClick={onHandleErrorClick}/>
        <GridRow>
          <GridCol>
            <BackLink onClick={e => this.goBack(e, backUrl)} href={backUrl}>
              {t('commonBackLinkBackButtonLabel')}
            </BackLink>
            <Header>{t('psAddHealthCertificateHeader')}</Header>
          </GridCol>
        </GridRow>
        <Form
          action="/orchestration/api/v1/processingStatement/saveAndValidate"
          currentUrl={path}
          nextUrl={nextUri}
          onSubmit={e => e.preventDefault()}>
          <GridRow>
            <GridCol columnTwoThirds>
              <InputField htmlFor="healthCertificateNumber" meta={{ error: t(errors.healthCertificateNumber), touched: true }} hint={t('psAddHealthCertificateHealthCertificateNumberHint')} input={{autoComplete: 'off', name: 'healthCertificateNumber', id: 'healthCertificateNumber', value: processingStatement.healthCertificateNumber, onChange: e => this.onChange(e)}}>{t('psAddHealthCertificateHealthCertificateNumber')}</InputField>
            </GridCol>
          </GridRow>
          <GridRow>
            <GridCol columnTwoThirds>
            <DateFieldWithPicker
                id='healthCertificateDate'
                name='healthCertificateDate'
                errors={t(errors.healthCertificateDate)}
                onDateChange={e => this.onChange(e)}
                dateFormat='DD/MM/YYYY'
                date={certificateDateString || ''}
                labelText={t('psAddHealthCertificateHealthCertificateDateLabel')}
            ></DateFieldWithPicker>
            </GridCol>
          </GridRow>
          <GridRow>
            <SaveAsDraftButton formactionUrl={saveAsDraftFormActionUrl} onClick={this.onSaveAsDraft} />
            <input type="hidden" name="journey" value={journey} />
            <input type="hidden" name="currentUri" value={path} />
            <ContinueButton id="continue" onClick={() => this.save(nextUri)}>
              {t('commonContinueButtonSaveAndContinueButton')}
            </ContinueButton>
          </GridRow>
        </Form>
        <BackToProgressLink
          progressUri={progressUri}
          documentNumber={documentNumber}
        />
        <HelpLink journey={route.journey}/>
      </Main>
    );
  }
}

function mapStateToProps(state) {
  return {
    processingStatement: state.processingStatement
  };
}

function loadData(store) {
  return store.dispatch(getProcessingStatementFromRedis(this.documentNumber));
}

export const component = withRouter(
  connect(mapStateToProps,
    {
      save: saveProcessingStatement,
      clear: clearProcessingStatement,
      saveToRedis: saveProcessingStatementToRedis,
      getFromRedis: getProcessingStatementFromRedis
    })(withTranslation()(AddHealthCertificatePage)));

export default {
  loadData,
  component: PageTemplateWrapper(component)
};
