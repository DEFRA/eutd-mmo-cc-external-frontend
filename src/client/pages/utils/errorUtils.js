import i18n from '../../../i18n';
import enTranslations from '../../../locales/uk/en.json';

export function toGovukErrors(obj) {
  return Object.keys( obj).map( o => {
    return {targetName: o, text: obj[o], key: o, message: (obj[o] in enTranslations) ? i18n.t(obj[o]) : obj[o]};
  });
}

export function onHandleErrorClick(targetName, speciesRow, vesselIndex) {
  let elem;

  if (speciesRow !== undefined && vesselIndex !== undefined) {
    elem = document.getElementById(targetName + '-' + speciesRow + '-' + vesselIndex);
  } else {
    elem = document.getElementsByName(targetName)[0] || document.getElementById(targetName);
  }

  if (elem) {
    elem.scrollIntoView({ behavior: 'smooth' });
    elem.focus({ preventScroll: true});
  }
}

export function onHandleErrorClickWithoutFocus(targetName) {
  const elem = document.getElementsByName(targetName)[0] || document.getElementById(targetName);
  elem.scrollIntoView({ behavior: 'smooth' });
}

export function scrollToErrorIsland() {
  setTimeout( () => {
    document.getElementById('errorIsland').scrollIntoView({behavior: 'smooth'});
    document.getElementById('errorIsland').focus({ preventScroll: true});
  },100);
}

export function scrollToField(id) {
  setTimeout( () => {
    document.getElementById(id).scrollIntoView({behavior: 'smooth'});
    document.getElementById(id).focus({ preventScroll: true});
  },100);
}

export function scrollToFieldName(name) {
  setTimeout( () => {
    document.getElementsByName(name)[0].scrollIntoView({behavior: 'smooth'});
    document.getElementsByName(name)[0].focus({ preventScroll: true});
  },100);
}

