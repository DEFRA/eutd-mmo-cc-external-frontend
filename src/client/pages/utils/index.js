import React, {Fragment} from 'react';

export * from './errorUtils';


export function formatAddress( ...adresssLines) {
  return adresssLines.filter( a => a ).map( (a, index) => <Fragment key={index}>{a}<br/></Fragment> );
}

export function constructAddress(addressArray) {
  let address = '';
  if (addressArray.length === 4) {
    if (addressArray[0]) {
      address += addressArray[0];
    }
    if (addressArray[0] && addressArray[1]) {
      address += ' ';
    }
    if (addressArray[1]) {
      address += addressArray[1];
    }
    if (addressArray[2]) {
      address += ', ' + addressArray[2];
    }
    if (addressArray[3]) {
      address += '. ' + addressArray[3];
    }
  }
  return address;
}

export const hasAdminOverride = (exportPayload) => {
  const isLandingOverridden = (landing) =>
    landing &&
    landing.model &&
    landing.model.vessel &&
    landing.model.vessel.vesselOverriddenByAdmin;

  return exportPayload &&
    exportPayload.items &&
    exportPayload.items.some(
      (item) =>
        item.landings &&
        item.landings.some((landing) => isLandingOverridden(landing))
    )
    ? true
    : false;
};