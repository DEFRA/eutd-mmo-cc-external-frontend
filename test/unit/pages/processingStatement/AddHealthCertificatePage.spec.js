import * as React from 'react';
import { mount } from 'enzyme/build';
import { Provider } from 'react-redux';
import { Router, Route } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import { createMemoryHistory } from 'history';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { getProcessingStatementFromRedis, clearProcessingStatement } from '../../../../src/client/actions';
import { component as AddHealthCertificate } from '../../../../src/client/pages/processingStatement/addHealthCertificatePage';
import AddHealthCertificatePageWrapper from '../../../../src/client/pages/processingStatement/addHealthCertificatePage';
import * as clientAction from '../../../../src/client/actions/index.js';
import * as ErrorUtils from '../../../../src/client/pages/utils/errorUtils';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

jest.mock('../../../../src/client/actions');

describe('Add health certificate page', () => {
  const mockStore = configureStore([thunk]);

  getProcessingStatementFromRedis.mockReturnValue({
    type: 'GET_PROCESSING_STATEMENT_FROM_REDIS',
  });

  it('should redirect to the previous page when required data is missing', async () => {
    const documentNumber = 'document123';
    const history = createMemoryHistory({
      initialEntries: [
        `/create-processing-statement/${documentNumber}/add-health-certificate`,
      ],
    });
    const mockPush = jest.spyOn(history, 'push');
    mockPush.mockReturnValue(null);

    const store = mockStore({
      processingStatement: {
        catches: [],
        consignmentDescription: 'consignment'
      },
      global: {
        allFish: []
      },
      clear: jest.fn
    });

    await act(async () => {
      await mount(
        <Provider store={store}>
          <Router history={history}>
            <Route path="/create-processing-statement/:documentNumber/add-health-certificate">
              <AddHealthCertificate
                route={{
                  previousUri:
                    '/create-processing-statement/:documentNumber/previous-uri',
                  path: 'path',
                  progressUri: '/create-processing-statement/:documentNumber/progress',
                  nextUri: 'nextUri'
                }}
              />
            </Route>
          </Router>
        </Provider>
      );
    });

    expect(mockPush).not.toHaveBeenCalled();
  });

  it('should call saveToRedis on save', async () => {
    const documentNumber = 'document123';
    const history = createMemoryHistory({
      initialEntries: [
        `/create-processing-statement/${documentNumber}/add-health-certificate`,
      ],
    });
    const mockPush = jest.spyOn(history, 'push');
    mockPush.mockReturnValue(null);

    const mockSaveToRedis = jest.spyOn(clientAction, 'saveProcessingStatementToRedis');

    const props = {
      saveToRedis: mockSaveToRedis,
      processingStatement: {
        catches: [],
        consignmentDescription: 'consignment',
        errors: {
          healthCertificateDate: "psAddHealthCertificateErrorRealDateHealthCertificateDate",
          healthCertificateNumber: "psAddHealthCertificateErrorFormatHealthCertificateNumber"
        }
      }
    };


    const store = mockStore({
      processingStatement: {
        catches: [],
        consignmentDescription: 'consignment',
        errors: {
          healthCertificateDate: "psAddHealthCertificateErrorRealDateHealthCertificateDate",
          healthCertificateNumber: "psAddHealthCertificateErrorFormatHealthCertificateNumber"
        }
      }
    });

    const wrapper = mount(
        <Provider store={store}>
          <Router history={history}>
            <Route path="/create-processing-statement/:documentNumber/add-health-certificate">
              <AddHealthCertificate
                route={{
                  previousUri:
                    '/create-processing-statement/:documentNumber/previous-uri',
                  progressUri: '/create-processing-statement/:documentNumber/progress',
                  path: 'path',
                  nextUri: 'nextUri',
                }}
                props = {props}
              />
            </Route>
          </Router>
        </Provider>
      );

    const continueBtn = wrapper.find('#continue');
    await continueBtn.at(0).simulate('click', {preventDefault: () => {}});

    expect(mockSaveToRedis).toHaveBeenCalled();
    expect(mockSaveToRedis).toHaveBeenCalledWith({
      data: props.processingStatement || {},
      currentUrl: '/create-processing-statement/document123/add-health-certificate',
      saveAsDraft: false,
      documentNumber: documentNumber,
      saveToRedisIfErrors: true
    });
  });

  it('should clear errors when component unmounts', () => {
    const mockDispatchClearErrors = jest.fn();

    new AddHealthCertificate.WrappedComponent({
      clear: mockDispatchClearErrors,
    }).componentWillUnmount();

    expect(mockDispatchClearErrors).toHaveBeenCalled();
  });

  it('should redirect to forbidden page when there is unauthorised access', () => {
    const mockPush = jest.fn();

    new AddHealthCertificate.WrappedComponent({
      match: {
        params: {
          documentNumber: 'doc1234',
        },
      },
      route: {
        previousUri: '/create-processing-statement/:documentNumber/previous-uri'
      },
      processingStatement: {
        unauthorised: true,
      },
      history: {
        push: mockPush,
      },
    }).componentDidUpdate();

    expect(mockPush).toHaveBeenCalledWith('/forbidden');
  });

  it('will call all methods needed to load the component', async () => {
    const store = {
      dispatch: () => {
        return new Promise((resolve) => {
          resolve();
        });
      }
    };

    await new AddHealthCertificatePageWrapper.loadData(store, 'processingStatement');

    expect(getProcessingStatementFromRedis).toHaveBeenCalled();
  });
});

describe('#Add health certificate page after translation', () => {
  const mockStore = configureStore([thunk]);
  const mockSaveToRedis = jest.spyOn(clientAction, 'saveProcessingStatementToRedis');
  const mockScrollToErrorIsland = jest.spyOn(ErrorUtils, 'scrollToErrorIsland');
  mockScrollToErrorIsland.mockReturnValue(null);

  getProcessingStatementFromRedis.mockReturnValue({
    type: 'GET_PROCESSING_STATEMENT_FROM_REDIS',
  });

  const documentNumber = 'document123';
  const history = createMemoryHistory({
    initialEntries: [
      `/create-processing-statement/${documentNumber}/add-health-certificate`,
    ],
  });
  const mockPush = jest.spyOn(history, 'push');
  mockPush.mockReturnValue(null);

  const getComponentWrapper = (store={}, props={}) => {
    const defaultStore = {
      processingStatement: {
        catches: [],
        consignmentDescription: 'consignment',
        unauthorised: true,
        errors: {
          healthCertificateDate: "psAddHealthCertificateErrorHealthCertificateDate",
          healthCertificateNumber: "psAddHealthCertificateErrorFormatHealthCertificateNumber"
        },
        errorsUrl: `/create-processing-statement/${documentNumber}/add-health-certificate`
      },
      global: {
        allFish: []
      },
      history: {
        location: {
          pathname: `/create-processing-statement/${documentNumber}/add-health-certificate`
        }
      },
      clear: jest.fn(),
      ...store
    };

    const _store = mockStore(defaultStore);

    const _props = {
      processingStatement: {
        catches: [],
        consignmentDescription: 'consignment',
        errors: {
          healthCertificateDate: "psAddHealthCertificateErrorRealDateHealthCertificateDate",
          healthCertificateNumber: "psAddHealthCertificateErrorFormatHealthCertificateNumber"
        }
      },
      match:{
        params: {
          documentNumber
        }
      },
      route: {
        previousUri:
          '/create-processing-statement/:documentNumber/previous-uri',
        path: 'path',
        progressUri: '/create-processing-statement/:documentNumber/progress',
        nextUri: 'nextUri'
      },
      history: {
        push: jest.fn
      },
      saveToRedis: mockSaveToRedis,
      ...props
    }

    return mount(
      <Provider store={_store}>
        <Router history={history}>
            <AddHealthCertificate
              route={{
                previousUri:
                  '/create-processing-statement/:documentNumber/previous-uri',
                path: 'path',
                progressUri: '/create-processing-statement/:documentNumber/progress',
                nextUri: 'nextUri'
              }}
              {..._props}
            />
        </Router>
      </Provider>
    );
  }

  it('will call the the on submit for component', () => {
    const wrapper = getComponentWrapper();
    const event = {
      preventDefault: jest.fn(),
    };
    const formEl = wrapper.find('form');
    formEl.props().onSubmit(event);

    expect(formEl.exists()).toBeTruthy();
    expect(event.preventDefault).toHaveBeenCalled();
  });

  it('should call isUnauthorised if unauthorised is true', () => {
    const wrapper = getComponentWrapper();
    expect(wrapper.find('AddHealthCertificate')).toBeDefined();
    expect(mockPush).toHaveBeenCalled();
  });

  it('should be scroll to errorIsLand with empty form submission', () => {
    const wrapper = getComponentWrapper();
    const event = {
      preventDefault: jest.fn(),
    };
    const formEl = wrapper.find('form');
    formEl.props().onSubmit(event);

    expect(wrapper.find('ul.error-summary-list li a').at(0).text()).toBe('Enter the Export Health Certificate date');
    expect(wrapper.find('ul.error-summary-list li a').at(1).text()).toBe('Enter Export Health Certificate number in the correct format');
  });

  it('should be scroll to errorIsLand with other errors', () => {
    const store = {
      processingStatement: {
        catches: [],
        consignmentDescription: 'consignment',
        unauthorised: true,
        errors: {
          healthCertificateDate: "psAddHealthCertificateErrorHealthCertificateDate",
          healthCertificateNumber: "psAddHealthCertificateErrorFormatHealthCertificateNumber"
        },
        errorsUrl: `/create-processing-statement/${documentNumber}/add-health-certificate`,
        healthCertificateDate: new Date((new Date()).getTime() + (10 * 86400000)).toLocaleDateString()
      }
    };
    const wrapper = getComponentWrapper(store);
    const event = {
      preventDefault: jest.fn(),
    };
    const formEl = wrapper.find('form');
    formEl.props().onSubmit(event);

    expect(formEl.exists()).toBeTruthy();
    expect(event.preventDefault).toHaveBeenCalled();
    expect(wrapper.find('ul.error-summary-list li a').at(1).text()).toBe('Enter Export Health Certificate number in the correct format');
  });
});

describe('Snapshot of Add health certificate page', () => {
  const mockStore = configureStore([thunk]);

  clearProcessingStatement.mockReturnValue({
    type: 'CLEAR_PROCESSING_STATEMENT'
  });

  getProcessingStatementFromRedis.mockReturnValue({
    type: 'GET_PROCESSING_STATEMENT_FROM_REDIS',
  });

  it('should match the snapshot', () => {
    const documentNumber = 'document123';
    const history = createMemoryHistory({
      initialEntries: [
        `/create-processing-statement/${documentNumber}/add-health-certificate`,
      ],
    });

    const store = mockStore({
      processingStatement: {
        catches: [],
        consignmentDescription: 'consignment'
      },
      global: {
        allFish: []
      }
    });

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <Route path="/create-processing-statement/:documentNumber/add-health-certificate">
            <AddHealthCertificate
              route={{
                previousUri:
                  '/create-processing-statement/:documentNumber/previous-uri',
                path: 'path',
                progressUri: '/create-processing-statement/:documentNumber/progress',
                nextUri: 'nextUri'
              }}
            />
          </Route>
        </Router>
      </Provider>
    );

    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  });
});