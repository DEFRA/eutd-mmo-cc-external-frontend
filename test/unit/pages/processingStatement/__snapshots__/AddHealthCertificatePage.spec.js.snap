// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Add health certificate page should match the snapshot 1`] = `
<div>
  <div
    class="processing-statement css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/create-processing-statement/document123/previous-uri"
          >
            Back
          </a>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            Add health certificate details
          </h1>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/processingStatement/saveAndValidate?c=path&n=nextUri"
        method="POST"
      >
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <label
              class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
              for="healthCertificateNumber"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              >
                Export Health Certificate (EHC) number
              </span>
              <span
                class="e1vcssft0 css-1whl0n4-StyledHint-StyledHoc e4u1u2g0"
              >
                For example, 20/2/123456
              </span>
              
              <input
                autocomplete="off"
                class="e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
                id="healthCertificateNumber"
                name="healthCertificateNumber"
                type="text"
                value=""
              />
            </label>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <div
              class="date-field-with-picker"
              id="healthCertificateDate"
            >
              <div
                class="column-field-picker"
              >
                <div
                  class="date-field e1vcssft0 css-vav5a4-StyledContainer-StyledHoc e4u1u2g0"
                >
                  <span
                    class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                  >
                    <span
                      class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                      id="date-field-label-text"
                    >
                      Export Health Certificate (EHC) date
                    </span>
                  </span>
                  <span
                    class="e1vcssft0 css-1whl0n4-StyledHint-StyledHoc e4u1u2g0"
                  >
                    For example, 31 03 1980
                  </span>
                  
                  <div
                    class="css-bw34j1-StyledList e107rt5v2"
                  >
                    <label
                      class="e107rt5v1 css-iwpl6b-StyledLabel-StyledHoc-StyledLabel e4u1u2g0"
                    >
                      <span
                        class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                      >
                        Day
                      </span>
                      <input
                        class="css-lktfuz-StyledInput e107rt5v0"
                        name="dayInputName"
                        type="number"
                        value=""
                      />
                    </label>
                    <label
                      class="e107rt5v1 css-iwpl6b-StyledLabel-StyledHoc-StyledLabel e4u1u2g0"
                    >
                      <span
                        class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                      >
                        Month
                      </span>
                      <input
                        class="css-lktfuz-StyledInput e107rt5v0"
                        name="monthInputName"
                        type="number"
                        value=""
                      />
                    </label>
                    <label
                      class="e107rt5v1 css-17bpbza-StyledLabel-StyledHoc-StyledLabel e4u1u2g0"
                    >
                      <span
                        class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                      >
                        Year
                      </span>
                      <input
                        class="css-lktfuz-StyledInput e107rt5v0"
                        name="yearInputName"
                        type="number"
                        value=""
                      />
                    </label>
                  </div>
                </div>
                <div
                  class="date-picker "
                >
                  <button>
                    <img
                      alt="calendar"
                      src="/static/assets/images/calendar.png"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <button
            class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
            formaction="/orchestration/api/v1/processingStatement/saveAndValidate?c=path&n=nextUri&saveAsDraftUrl=undefined"
            id="saveAsDraft"
            name="saveAsDraft"
            type="submit"
          >
            Save as Draft
          </button>
          <input
            name="journey"
            type="hidden"
            value=""
          />
          <input
            name="currentUri"
            type="hidden"
            value="path"
          />
          <button
            class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
            id="continue"
          >
            Save and continue
          </button>
        </div>
      </form>
      <a
        href="/create-processing-statement/document123/progress"
        id="backToProgress"
      >
        Back to Your Progress
        <span
          class="govuk-visually-hidden"
        >
          back to your progress page link
        </span>
      </a>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
